src/libsystemd/sd-netlink/sd-netlink.c

```diff
@@ -227,6 +227,31 @@ int sd_netlink_send(sd_netlink *nl,
         return 1;
 }
 
+int sd_netlink_sendv(sd_netlink *nl,
+                     sd_netlink_message *messages[],
+                     unsigned int msgcount,
+                     uint32_t *serial) {
+        unsigned int i;
+        int r;
+
+        assert_return(nl, -EINVAL);
+        assert_return(!rtnl_pid_changed(nl), -ECHILD);
+        assert_return(messages, -EINVAL);
+
+        for (i = 0; i < msgcount; i++) {
+                assert_return(!messages[i]->sealed, -EPERM);
+                rtnl_seal_message(nl, messages[i]);
+                if (serial)
+                        serial[i] = rtnl_message_get_serial(messages[i]);
```

Comment:
as above, do not clobber return values on failure

Reply:
I've resolved this by moving the allocation of serial[ ] from the caller to this function. Caller now passes "uint32_t **serials"
instead of "*serial". **serial is only updated on success.